ugvc/filtering/training_prep.py

```diff
@@ -0,0 +1,304 @@
+            axis=1,
+        )
+
+        result_gt1[fp_ca_df.index] = gt_genotypes.apply(lambda x: x[0])
+        result_gt2[fp_ca_df.index] = gt_genotypes.apply(lambda x: x[1])
+    result = pd.DataFrame({0: result_gt1.astype(int), 1: result_gt2.astype(int)})
+    result["true_gt"] = result.apply(lambda x: (x[0], x[1]), axis=1)
+    return result["true_gt"]
+
+
+def prepare_ground_truth(
+    input_vcf: str, base_vcf: str, hcr: str, reference: str, output_h5: str, chromosome: list | None = None
+) -> None:
+    """Generates a training set dataframe from callset and the ground truth VCF. The following steps are peformed:
+    1. Run vcfeval to compare the callset to the ground truth
+    2. Join the callset and the vcfeval output
+    3. Calculate the labels for each variant
+    4. Process multiallelic variants and spanning deletions (splitting them into multiple lines,
+    asking each time if one or two non-reference alleles are present). For example if the true
+    variant is (1,2), we will generate two rows in the training df:
+    in the first row we will have the reference and the strongest
+    alt allele and the label is (1,1) and in the second row we will have
+    the strongest alt allele and the second strongest alt allele and the label will
+    be (0, 1)
+    The results are saved in output_h5 file, with each chromosome in a separate key.
+
+    Parameters
+    ----------
+    input_vcf : str
+        Input (call) VCF file
+    base_vcf : str
+        Truth VCF file (e.g GIAB callset)
+    hcr : str
+        HCR bed file to generate training set on
+    reference : str
+        Reference fasta. Note that .sdf file will be required
+    output_h5 : str
+        Output file
+    chromosome : list, optional
+        List of chromosomes to operate on, by default None
+
+    """
+    pipeline = vpu.VcfPipelineUtils()
+    vcfeval_output = pipeline.run_vcfeval_concordance(
+        input_file=input_vcf,
+        truth_file=base_vcf,
+        output_prefix=input_vcf.replace(".vcf.gz", ""),
+        ref_genome=reference,
+        evaluation_regions=hcr,
+        ignore_filter=True,
+    )
+    if chromosome is None:
+        chromosome = [f"chr{x}" for x in list(range(1, 23)) + ["X", "Y"]]
+
+    for chrom in tqdm.tqdm(chromosome):
+        labeled_df = calculate_labeled_vcf(input_vcf, vcfeval_output, contig=chrom)
+        labels = calculate_labels(labeled_df)
+        labeled_df["label"] = labels
+        labeled_df = process_multiallelic_spandel(labeled_df, reference, chrom, input_vcf)
+        labeled_df.to_hdf(output_h5, key=chrom, mode="a")
+
+
+def encode_labels(ll: Iterable[tuple[int, int]]) -> list[int]:
+    """Convert genotype label (tuple) to number
+
+    Parameters
+    ----------
+    ll : Iterable[tuple[int, int]]
+        List of labels
+
+    Returns
+    -------
+    list[int]
+        List of encoded labels
+    """
+    return [encode_label(x) for x in ll]
+
+
+def encode_label(label: tuple[int, ...]) -> int:
```

Comment:
Could (0,0): 0, (0,1): 1, (1,1): 2 be more straightforward?

Reply:
Yes, fixed in later PR :)